packages/block-library/src/query-pagination-numbers/edit.js

```diff
@@ -1,25 +1,73 @@
 /**
  * WordPress dependencies
  */
-import { useBlockProps } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { InspectorControls, useBlockProps } from '@wordpress/block-editor';
+import { PanelBody, RangeControl } from '@wordpress/components';
 
 const createPaginationItem = ( content, Tag = 'a', extraClass = '' ) => (
-	<Tag className={ `page-numbers ${ extraClass }` }>{ content }</Tag>
+	<Tag key={ content } className={ `page-numbers ${ extraClass }` }>
+		{ content }
+	</Tag>
 );
 
-const previewPaginationNumbers = () => (
-	<>
-		{ createPaginationItem( 1 ) }
-		{ createPaginationItem( 2 ) }
-		{ createPaginationItem( 3, 'span', 'current' ) }
-		{ createPaginationItem( 4 ) }
-		{ createPaginationItem( 5 ) }
-		{ createPaginationItem( '...', 'span', 'dots' ) }
-		{ createPaginationItem( 8 ) }
-	</>
-);
+const previewPaginationNumbers = ( midSize ) => {
+	const paginationItems = [];
+
+	// First set of pagination items.
+	for ( let i = 1; i <= midSize; i++ ) {
+		paginationItems.push( createPaginationItem( i ) );
+	}
+
+	// Current pagination item.
+	paginationItems.push(
+		createPaginationItem( midSize + 1, 'span', 'current' )
+	);
+
+	// Second set of pagination items.
+	for ( let i = 1; i <= midSize; i++ ) {
+		paginationItems.push( createPaginationItem( midSize + 1 + i ) );
+	}
+
+	// Dots.
+	paginationItems.push( createPaginationItem( '...', 'span', 'dots' ) );
+
+	// Last pagination item.
+	paginationItems.push( createPaginationItem( midSize * 2 + 2 ) );
```

Comment:
Just a tiny change here.. If there are just enough pages to show based on `mid_size`, the last `...` are not rendered. For example we render `1 2` and not `1...2`. So we could just add a bigger number here.
```suggestion
	paginationItems.push( createPaginationItem( midSize * 2 + 3 ) );
```

Reply:
Good point, I updated the code accordingly.

I also rebuilt the tests, but still some checks are failing. Do you know why this is the case?

Reply:
Oh, now they went through, great!